gen/gen.go

```diff
@@ -0,0 +1,154 @@
+	if err != nil {
+		return err
+	}
+
+	testTemplateFile := filepath.Join("gen", "templates", "integration_resource_test.go.tmpl")
+	testTmpl, err := template.ParseFiles(testTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	resourceDotTFTemplateFile := filepath.Join("gen", "templates", "resource.tf.tmpl")
+	resourceTFTmpl, err := template.ParseFiles(resourceDotTFTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the output directory exists
+	outputDirPath := filepath.Join("gen", "generated")
+	if err := os.MkdirAll(outputDirPath, 0755); err != nil {
+		return err
+	}
+
+	i := mondoov1.ClientIntegrationConfigurationInput{}
+	m := structs.Map(i)
+
+	for k := range m {
+		// TODO we know the type and the struct associated to the type, we need
+		// to look it (the struct) and use the same `structs.Map(v)` to list all
+		// fields per integration and auto generate the terraform schema and more
+		// details, for now, we only leave a comment where we need to add specific
+		// integration options
+
+		var (
+			className, _          = strings.CutSuffix(k, "ConfigurationOptions")
+			terraformResourceName = strings.ToLower(toSnakeCase(className))
+			resource              = IntegrationResource{
+				ResourceClassName:     className,
+				TerraformResourceName: terraformResourceName,
+			}
+		)
+
+		fmt.Printf("> Generating code for %s integration\n", className)
+
+		// Create the resource file
+		resourceOutputFilePath := filepath.Join(outputDirPath,
+			fmt.Sprintf("integration_%s_resource.go", terraformResourceName),
+		)
+		resourceFile, err := os.Create(resourceOutputFilePath)
+		if err != nil {
+			return err
+		}
+		defer resourceFile.Close()
+
+		if err := resourceTmpl.Execute(resourceFile, resource); err != nil {
+			return err
+		}
+
+		// Create test file
+		testOutputFilePath := filepath.Join(outputDirPath,
+			fmt.Sprintf("integration_%s_resource_test.go", terraformResourceName),
+		)
+		testFile, err := os.Create(testOutputFilePath)
+		if err != nil {
+			return err
+		}
+		defer testFile.Close()
+
+		if err := testTmpl.Execute(testFile, resource); err != nil {
+			return err
+		}
+
+		// Create examples/ files
+		examplesDirPath := filepath.Join(outputDirPath, "examples", fmt.Sprintf("mondoo_integration_%s", terraformResourceName))
+		// Ensure the output directory exists
+		if err := os.MkdirAll(examplesDirPath, 0755); err != nil {
+			return err
+		}
+		// Create example main.tf
+		err = os.WriteFile(filepath.Join(examplesDirPath, "main.tf"), mainDotTFTestFile(), 0644)
+		if err != nil {
+			return err
+		}
+		// Create example resource.tf
+		resourceTFOutputFilePath := filepath.Join(examplesDirPath, "resource.tf")
+		resourceTFFile, err := os.Create(resourceTFOutputFilePath)
+		if err != nil {
+			return err
+		}
+		defer testFile.Close()
+
+		if err := resourceTFTmpl.Execute(resourceTFFile, resource); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
```

Comment:
I'm not sure this works as expected. Wouldn't the `.` also match an upper case letter?

Reply:
I think, I'm missing something here. 
Could you please add a comment or a test case to show how this works?